src/main/java/mate/jdbc/lib/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+            throw new DataProcessingException("Can't insert manufacturer " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllFromManufacturers = "SELECT * FROM manufacturers WHERE is_deleted = FALSE;";
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        Manufacturer manufacturer;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement =
+                        connection.prepareStatement(getAllFromManufacturers)) {
+            ResultSet resultSet = getAllManufacturersStatement
+                    .executeQuery(getAllFromManufacturers);
+            while (resultSet.next()) {
+                manufacturer = getManufacturer(resultSet);
+                allManufacturers.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get \"all manufacturers\" from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getFromManufacturers = "SELECT * FROM manufacturers WHERE id = ?";
+        Manufacturer manufacturer;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturersStatement =
+                        connection.prepareStatement(getFromManufacturers)) {
+            getManufacturersStatement.setString(1, "" + id);
+            ResultSet resultSet = getManufacturersStatement.executeQuery();
+            manufacturer = getManufacturer(resultSet);
+            if (manufacturer.isDeleted()) {
+                throw new NoDataException("Can't get data from \"manufacturer\". "
+                        + "String with id = " + id + " was deleted");
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer by id " + id, e);
+        }
+        return Optional.ofNullable(manufacturer);
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacturersRequest = "UPDATE manufacturers SET "
+                + "name = ?, country = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturersStatement =
+                        connection.prepareStatement(updateManufacturersRequest)) {
+            updateManufacturersStatement.setString(1, manufacturer.getName());
+            updateManufacturersStatement.setString(2, manufacturer.getCountry());
+            updateManufacturersStatement.setString(3, "" + manufacturer.getId());
+            int i = updateManufacturersStatement.executeUpdate();
+            if (i == 0) {
+                throw new NoDataException("Can't update string with id = " + manufacturer.getId());
+            }
```

Comment:
```suggestion
            updateManufacturersStatement.executeUpdate();
```

This check is not needed. If something goes wrong, we will have an exception.

Reply:
Yes, I have tried, but I haven't got any exception without this check...

Reply:
I mean in case when you try to UPDATE non-existing string... For example, you have only 7 strings, and you are trying to update the 8th one.

Reply:
If you have any questions about comments you should discuss them in Slack, not here. Do not request new review before everything is clear.